scripts/make_tor_targets.py

```diff
@@ -0,0 +1,114 @@
+import json
+import hashlib
+
+from urllib.request import urlopen
+
+DIRAUTH_URL = "https://gitweb.torproject.org/tor.git/plain/src/app/config/auth_dirs.inc"
+BRIDGES_URL = "https://bridges.torproject.org/moat/circumvention/builtin"
+
+def parse_params(parts : list[str]) -> dict[str, str]:
+    params = {}
+    for p in parts:
+        k, v = p.split("=")
+        params[k] = [v]
+    return params
+
+def parse_bridge_line(line : str) -> tuple[str, dict]:
+    bridge = {}
+    parts = line.split(" ")
+    bridge["protocol"] = parts[0]
+    bridge["address"] = parts[1]
+    bridge["fingerprint"] = parts[2]
+    bridge["params"] = parse_params(parts[3:])
+
+    bridge_id = hashlib.sha256(
+            bridge["address"].encode("ascii") + bridge["fingerprint"].encode("utf-8")
+    ).hexdigest()
+
+    return bridge_id, bridge
+
+def get_bridges():
+    with urlopen(BRIDGES_URL) as resp:
+        j = json.loads(resp.read())
+
+    bridges = {}
+    for b in j["obfs4"]:
+        bridge_id, bd = parse_bridge_line(b)
+        assert bridge_id not in bridges
+        bridges[bridge_id] = bd
+    return bridges
+
+def parse_dirauth(line : str) -> dict:
```

Comment:
As before, it would be beneficial to add an example we can publicly share as a comment, such that whoever is reading the code in the future could more easily and immediately view that the code is correct.

```suggestion
def parse_dirauth(line : str) -> dict:
    # Example: tor26 orport=443 v3ident=14C131DFC5C6F93646BE72FA1401C02A8DF2E8B4 ipv6=[2001:858:2:2:aabb:0:563b:1526]:443 86.59.21.38:80 847B 1F85 0344 D787 6491 A548 92F9 0493 4E4E B85D
```

Again, the example is taken from the public API and so I'm comfortable adding it.